test/support/utils.go

```diff
@@ -36,6 +36,24 @@ const (
 
 func WriteToOutputDir(t Test, fileName string, fileType OutputType, data []byte) {
 	t.T().Helper()
-	t.Expect(os.WriteFile(path.Join(t.OutputDir(), fileName+"."+string(fileType)), data, fs.ModePerm)).
-		To(gomega.Succeed())
+
+	// Get output directory and create the directory if it doesn't exist
+	outputDir := t.OutputDir()
+	err := os.MkdirAll(outputDir, os.ModePerm)
```

Comment:
If we _really_ want to have the parent directory created, it'd be better to encapsulate the logic in the `OutputDir()` method. Still the existing assumption is that the parent directory exist, and the output directory is created there.

Reply:
Thanks @astefanutti ,
For some reason the /tmp/TestMNISTRayJobMCADRayCluster directory isn't being created before that, so the logs couldn't be saved.

Reply:
In what scenario did it fail? Is it when you run the test locally?

Reply:
Yea, I've been running the codeflare-operator locally via openshift

Reply:
OK, so when run locally, the `CODEFLARE_TEST_OUTPUT_DIR` environment variable is supposed to be used. It's somehow designed so as to avoid having the test execution polluting the disk by mistake.

Reply:
Very Nice :D Thanks @astefanutti  these changes have been pushed.